tests/std/tests/VSO_0000000_regex_use/test.cpp

```diff
@@ -1698,6 +1698,190 @@ void test_gh_5377() {
+            plus_quantifier_regex_ecma_or_ere.should_search_match_capture_groups("b", "", match_default, {{0, 0}});
+            plus_quantifier_regex_ecma_or_ere.should_search_match_capture_groups("a", "a", match_default, {{0, 1}});
+        }
+    }
+
+    for (auto option : {basic, grep}) {
+        test_regex plus_quantifier_regex_bre(&g_regexTester, R"(\(a*\)\{1,\})", option);
+        plus_quantifier_regex_bre.should_search_match_capture_groups("", "", match_default, {{0, 0}});
+        plus_quantifier_regex_bre.should_search_match_capture_groups("b", "", match_default, {{0, 0}});
+        plus_quantifier_regex_bre.should_search_match_capture_groups("a", "a", match_default, {{0, 1}});
+    }
+
+    for (string pattern : {"(a*){1}", "(a?){1}"}) {
+        for (auto option : {ECMAScript, extended, egrep, awk}) {
+            test_regex repeat_once_regex_ecma_or_ere(&g_regexTester, pattern, option);
+            repeat_once_regex_ecma_or_ere.should_search_match_capture_groups("", "", match_default, {{0, 0}});
+            repeat_once_regex_ecma_or_ere.should_search_match_capture_groups("b", "", match_default, {{0, 0}});
+            repeat_once_regex_ecma_or_ere.should_search_match_capture_groups("a", "a", match_default, {{0, 1}});
+        }
+    }
+
+    for (string pattern : {R"(\(a*\)\{1\})", R"(\(a\{0,1\}\)\{1\})"}) {
+        for (auto option : {basic, grep}) {
+            test_regex repeat_once_regex_bre(&g_regexTester, R"(\(a*\)\{1\})", option);
+            repeat_once_regex_bre.should_search_match_capture_groups("", "", match_default, {{0, 0}});
+            repeat_once_regex_bre.should_search_match_capture_groups("b", "", match_default, {{0, 0}});
+            // leftmost-longest rule according to Boost semantics
+            repeat_once_regex_bre.should_search_match_capture_groups("a", "a", match_default, {{0, 1}});
+        }
+    }
+
+    for (string pattern : {"(a*){2}", "(a?){2}"}) {
+        test_regex repeat_twice_regex_ecma(&g_regexTester, pattern, ECMAScript);
+        repeat_twice_regex_ecma.should_search_match_capture_groups("", "", match_default, {{0, 0}});
+        repeat_twice_regex_ecma.should_search_match_capture_groups("b", "", match_default, {{0, 0}});
+        repeat_twice_regex_ecma.should_search_match_capture_groups("a", "a", match_default, {{1, 1}});
+
+        for (auto option : {extended, egrep, awk}) {
+            test_regex repeat_twice_regex_ere(&g_regexTester, pattern, option);
+            repeat_twice_regex_ere.should_search_match_capture_groups("", "", match_default, {{0, 0}});
+            repeat_twice_regex_ere.should_search_match_capture_groups("b", "", match_default, {{0, 0}});
+            // leftmost-longest rule according to Boost semantics
+            repeat_twice_regex_ere.should_search_match_capture_groups("a", "a", match_default, {{0, 1}});
+        }
+    }
+
+    for (string pattern : {R"(\(a\{0,1\}\)\{2\})", R"(\(a*\)\{2\})"}) {

```

Comment:
Nitpick: When "translating" ECMA/ERE syntax to BRE, let's list the patterns in the same order so the parallel structure is clearer.